builder/evaluator.go

```diff
@@ -136,7 +138,13 @@ func (b *Builder) Run(context io.Reader) (string, error) {
 		}
 	}()
 
-	filename := path.Join(b.contextPath, "Dockerfile")
+	filename := path.Join(b.contextPath, b.dockerfileName)
+
+	absContextPath, _ := filepath.Abs(b.contextPath)
```

Comment:
Not sure what I was thinking at the time other than it seemed pretty unlikely that they would fail but I'll add 'em.